### PR TITLE
[WIP] Extract all external function declarations to headers

### PR DESCRIPTION
**Pull request by marijnvdwerf**
Based on #111

### CODE REVIEW

File: src/pokemon_2.c [hunk 1]
```diff
@@ -50,12 +50,6 @@ extern u8 gUnknown_0820823C[];
 extern u8 gStatStageRatios[][2];
 extern u8 gHoldEffectToType[][2];
 
-extern u8 battle_side_get_owner(u8);
-extern u8 battle_get_side_with_given_state(u8);
-extern u32 battle_get_per_side_status(u8);
-extern u8 sub_8018324(u8, u8, u8, u8, u16);
-extern u8 sub_803C348(u8);
-
 u8 sub_803C348(u8 a1)
```
Comment (marijnvdwerf) [inline src/pokemon_2.c]: Has two errors in the generated pokemon_2.s:

(left is `master`, right is this PR)
image
image

**Comment by marijnvdwerf** [comment]
This gets rid of all `implicit declaration of function '{{ name}}'` warnings.

**Comment by marijnvdwerf** [comment]
Two issues remaining in `sub_803C434 `:
image
image

**Comment by YamaArashi** [comment]
Yeah. That function needed a lot of hacks to get it to match. I suspect there wouldn't be a problem if the function were written correctly. I'm not sure if there's any solution but to change the function back to asm.

**Pull request merged**